File: perl/hash.pl
# Author: David Santiago <https://github.com/demanuel/dessubdb>
# Excerpt from the project dessubdb

# Calculates the file hash, as specified in the thesubdb site
sub getFileHash {
      my $filename = shift or die 'A filename is required!';
      my $filesize = -s $filename;
      die "The file is too small! Wrong file?!? " unless $filesize > BLOCK_SIZE*2;
      open (my $fh, "<", $filename) or die "Couldn't open file $filename: $!";
      binmode $fh;
      my $buffer;
      my $data;
      read $fh, $data, BLOCK_SIZE;
      $buffer = $data;
      seek $fh, -1*BLOCK_SIZE, 2;
      read $fh, $data, BLOCK_SIZE;
      $buffer .=$data;
      close $fh;
      say "Hash= ".md5_hex($buffer);
      md5_hex($buffer);
}

#Usage

print getFileHash($filename);

File: php/hash.php
/**
 * Author: Raphael Santiago <https://github.com/RafaelSantiago>
 * Excerpt from the project php-subdb
 */

function hashSubDB($filename){
    $size = filesize($filename);
    $inicio = file_get_contents($filename, false, null, 0, 64 * 1024);
    $fim = file_get_contents($filename, false, null, $size - (64 * 1024), 64 * 1024);
    $data = $inicio . $fim;
    $hash = md5($data);
    return $hash;
}
